sdk-lib/src/main/java/cash/z/ecc/android/sdk/model/FastestServersResult.kt

```diff
@@ -0,0 +1,5 @@
+package cash.z.ecc.android.sdk.model
+
+import co.electriccoin.lightwallet.client.model.LightWalletEndpoint
+
+data class FastestServersResult(val servers: List<LightWalletEndpoint>?, val isLoading: Boolean)
```

Comment:
Wouldn't it make sense to create this sealed class and let it hold states like:
- Loading, SortedByLatency(servers), and SortedBySyncingBlocks(servers)
- or Loading, Processing(servers), and Sorted(servers)
- or omit the middle step's servers parameter completely?
Either way, we'd avoid having the servers nullable.

Reply:
This data structure is simpler. Because it is basically a cartesian product of all possible states.

isLoading=true & servers = null -> not loaded yet, but loading in progress
isLoading=true & servers != null -> loaded already & loading in progress to fetch newer data <- this one is very important
isLoading=false & servers = null -> not loaded, not loading
isLoading=false & servers != null -> loaded already & not fetching newer data

Reply:
also servers is always sorted :)

Reply:
You described exactly why calculated states (sealed class) might be better for the public API here. We don't document the state calculation anywhere; thus, calculating the API state seems reasonable here (of course, with all the necessary parameters in each state).